mp/src/game/server/pointteleport.cpp

```diff
@@ -145,3 +147,41 @@ void CPointTeleport::InputTeleport( inputdata_t &inputdata )
 	pTarget->Teleport( &m_vSaveOrigin, &m_vSaveAngles, NULL );
 }
 
+//------------------------------------------------------------------------------
+// Purpose: Teleport the entity given in the input parameter
+//------------------------------------------------------------------------------
+void CPointTeleport::InputTeleportEntity(inputdata_t &inputdata)
+{
+    // Attempt to find the entity in question
+    CBaseEntity *pTarget = gEntList.FindEntityByName(NULL, inputdata.value.String(), this, inputdata.pActivator, inputdata.pCaller);
+    if (pTarget == NULL)
+        return;
+
+    // If teleport object is in a movement hierarchy, remove it first
+    if (EntityMayTeleport(pTarget) == false)
+    {
+        Warning("ERROR: (%s) can't teleport object (%s) as it has a parent (%s)!\n", GetDebugName(),
+                pTarget->GetDebugName(), pTarget->GetMoveParent()->GetDebugName());
+        return;
+    }
+
+    // in episodic, we have a special spawn flag that forces Gordon into a duck
+#ifdef HL2_EPISODIC
+    if ((m_spawnflags & SF_TELEPORT_INTO_DUCK) && pTarget->IsPlayer())
+    {
+        CBasePlayer *pPlayer = ToBasePlayer(pTarget);
+        if (pPlayer != NULL)
+        {
+            pPlayer->m_nButtons |= IN_DUCK;
+            pPlayer->AddFlag(FL_DUCKING);
+            pPlayer->m_Local.m_bDucked = true;
+            pPlayer->m_Local.m_bDucking = true;
+            pPlayer->m_Local.m_flDucktime = 0.0f;
+            pPlayer->SetViewOffset(VEC_DUCK_VIEW_SCALED(pPlayer));
+            pPlayer->SetCollisionBounds(VEC_DUCK_HULL_MIN, VEC_DUCK_HULL_MAX);
+        }
+    }
+#endif
+
+    pTarget->Teleport(&m_vSaveOrigin, &m_vSaveAngles, NULL);
+}
```

Comment:
This teleport code looks very identical to the other teleport method, pull the shared bits out into a private internal method and call it from both.